src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -250,7 +250,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             End If
 
             If compareKind = TypeCompareKind.ConsiderEverything Then
-                Return t1.Equals(t2)
+                Return t1.Equals(DirectCast(t2, Object))
```

Comment:
>DirectCast(t2, Object) [](start = 33, length = 22)

Is the cast necessary?

Reply:
Yes, there are two overloads for equals (ISymbol) and (Object). Because t2 is a TypeSymbol we'll end up calling the ISymbol overload if we don't cast, which just infinitely recurses. Is there a better way of casting in VB?

---
In reply to: [305553202](https://github.com/dotnet/roslyn/pull/37247#discussion_r305553202) [](ancestors = 305553202)

Reply:
I see, thanks. That seems like another reason to make the `ISymbol.Equals()` implementation `Private`.
```
Private Function ISymbol_Equals(other As ISymbol)
    As Boolean Implements IEquatable(Of ISymbol).Equals
```

---
In reply to: [305962066](https://github.com/dotnet/roslyn/pull/37247#discussion_r305962066) [](ancestors = 305962066,305553202)